server/app/services/task-service.js

```diff
@@ -0,0 +1,173 @@
+const logger = require('../../logger');
+const { Barman, Kommission, Task } = require('../models');
+const { createUserError, createServerError, cleanObject, setEmbeddedAssociations } = require('../../utils');
+const sequelize = require('../../db');
+
+/**
+ * Create a task.
+ *
+ * @param newTask {Task} partial task
+ * @param _embedded {Object} Object containing associations to update, see swagger for more information.
+ * @return {Promise<Task|Errors.ValidationError>} The created task with its id
+ */
+async function createTask(newTask, _embedded) {
+
+    logger.verbose('Task service: creating a new task named %s', newTask.name);
+
+    const transaction = await sequelize.transaction();
+    try {
+        newTask.kommissionId = _embedded.kommissionId;
+        delete _embedded.kommissionId;
+        await newTask.save({ transaction });
+    } catch (err) {
+        logger.warn('Task service: Error while creating task', err);
+        await transaction.rollback();
+        throw createServerError('ServerError', 'Error while creating task');
+    }
+
+    //Associations
+    if (_embedded) {
+        for (const associationKey of Object.keys(_embedded)) {
+            const value = _embedded[associationKey];
+
+            await setEmbeddedAssociations(associationKey, value, newTask, transaction, true);
```

Comment:
Here, a barman could be assigned even if he is not in the kommission 
Should we check or not?

Reply:
Yes we schould.
As I said to @ClementGuittat , we could define a new permission called `allTask:write`. As a result, a user could write a task to any kommission even if he doesn't belong to it.

What do you think ?

Reply:
We could do something like this indeed, but it's hard to implement because the service doesn't know the user permissions (only the controller)...

We have to create another service to check if a barman is in the same kommissions as the task...

It's hard, and I don't kow how to handle Special Accounts

For now I think anyone can assign anybody and we will see later ?

Reply:
Ok, let's say that every barman can edit task beside his/her kommission.

But, if we want to be more fexlible (I mean, if some barman don't have the permission to edit task), we schould implement my way:

Define `allTask:write` permission and use a guard to check it as we currently do with other permissions.
We schould add this permission to the new barmanHEAD role !

Reply:
I don't really understand, You want to create a guard which will check if the user has the permissions allTask:write, but when do you want to check ?

Because you will still need to know if the user is in a Kommission (in case he does not have the perm) No ?

Reply:
Ok, so for now, every barman could create and update tasks !